packages/components/input/src/input.vue

```diff
@@ -324,9 +324,21 @@ const resizeTextarea = () => {
   if (autosize) {
     const minRows = isObject(autosize) ? autosize.minRows : undefined
     const maxRows = isObject(autosize) ? autosize.maxRows : undefined
+    const textareaStyle = calcTextareaHeight(textarea.value!, minRows, maxRows)
+
+    // If the scrollbar is displayed, the height of the textarea needs more space than the calculated height.
+    // If set textarea height in this case, the scrollbar will not hide.
+    // So we need to hide scrollbar first, and reset it in next tick.
+    // see https://github.com/element-plus/element-plus/issues/8825
     textareaCalcStyle.value = {
-      ...calcTextareaHeight(textarea.value!, minRows, maxRows),
+      overflowY: 'hidden',
+      ...textareaStyle,
     }
+
+    nextTick(() => {
+      textarea.value!.offsetHeight
```

Comment:
**Question:** may I ask what is the point of this line of code here?

Reply:
> **Question:** may I ask what is the point of this line of code here?

Because component update Effect and nextTick callback function are in the same Microtask Queue, so have to force the browser to repaint.

Here is `nextTick` implement:

```
export function nextTick<T = void>(
  this: T,
  fn?: (this: T) => void
): Promise<void> {
  const p = currentFlushPromise || resolvedPromise
  return fn ? p.then(this ? fn.bind(this) : fn) : p
}

```

It works like:

```
Promise.resolve().then(function () {
  el.setAttribute('style', 'height: 100px; overflow: hidden;')
}).then(function () {
  el.setAttribute('style', 'height: 100px;')
})
```

May be not use `nextTick`，use `setTimeout` is better?

Reply:
I prefer to use force repaint. Because `setTimeout` is async, it may cause some problems in the edge case.

here is force repaint in 6x slower cpu:
![20221019194148](https://user-images.githubusercontent.com/10278227/196681222-d6133e20-c7e9-4215-8aaa-7264849701a2.gif)

here is setTimeout in 6x slower cpu
![20221019193940](https://user-images.githubusercontent.com/10278227/196680838-06f8f774-46c0-4dd3-89fa-5f3fe79a500e.gif)